types/react-leaflet/index.d.ts

```diff
@@ -1,235 +1,351 @@
-// Type definitions for react-leaflet 1.0
+// Type definitions for react-leaflet 1.1
 // Project: https://github.com/PaulLeCam/react-leaflet
-// Definitions by: Dave Leaver <https://github.com/danzel>
+// Definitions by: Dave Leaver <https://github.com/danzel>, David Schneider <https://github.com/davschne>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.1
+// TypeScript Version: 2.2
 
 import * as Leaflet from 'leaflet';
 import * as React from 'react';
 
 // All events need to be lowercase so they don't collide with React.DOMAttributes<T>
 // which already declares things with some of the same names
 
-export interface LeafletLayerEvents {
-    onbaselayerchange?(event: Leaflet.LayersControlEvent): void;
-    onoverlayadd?(event: Leaflet.LayersControlEvent): void;
-    onoverlayremove?(event: Leaflet.LayersControlEvent): void;
-    onlayeradd?(event: Leaflet.LayerEvent): void;
-    onlayerremove?(event: Leaflet.LayerEvent): void;
-}
-export interface LeafletMapStateChangeEvents {
-    onzoomlevelschange?(event: Leaflet.Event): void;
-    onresize?(event: Leaflet.ResizeEvent): void;
+type Children = React.ReactNode | React.ReactNode[];
```

Comment:
Hi, @danzel, thanks for your prompt feedback. I based this directly on the `children` type defined in react-leaflet. See: [children.js](https://github.com/PaulLeCam/react-leaflet/blob/master/src/propTypes/children.js). React, for its part, seems to allow absolutely anything to be passed as the `children` prop. I agree it would be great to enforce only children that Leaflet allows, but I think that could be a lot of work to implement. And since react-leaflet doesn't enforce such a restriction, either via Flow or React propTypes, it seems like going beyond the call of duty to do so here.